lib/components/src/Accordion/AccordionHeader.tsx

```diff
@@ -0,0 +1,199 @@
+import { styled } from '@storybook/theming';
+import React, { FC, useContext, useEffect, useRef, useState } from 'react';
+import { Icons } from '../icon/icon';
+import { AccordionItemContext } from './AccordionItemContext';
+
+export type AccordionHeaderProps = {
+  label?: string;
+  hideIcon?: boolean;
+  Icon?: React.ReactNode;
```

Comment:
This should just be 1 prop: `icon` (lowercase, because it's a Node, not a Component).

If the prop is not provided, the "slot" isn't rendered.
I don't thin it makes sense, to provide a component with data AND instructions not to show said data.

Reply:
icon vs Icon is so that you can either just use our own baked in icons (and icon component) or your own icon or icon component.